airbyte-cdk/python/airbyte_cdk/sources/utils/slice_logger.py

```diff
@@ -0,0 +1,49 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import json
+import logging
+from abc import ABC, abstractmethod
+from typing import Any, Mapping, Optional
+
+from airbyte_cdk.models import AirbyteLogMessage, AirbyteMessage, Level
+from airbyte_cdk.models import Type as MessageType
+
+
+class SliceLogger(ABC):
+    SLICE_LOG_PREFIX = "slice:"
+
+    def create_slice_log_message(self, _slice: Optional[Mapping[str, Any]]) -> AirbyteMessage:
+        """
+        Mapping is an interface that can be implemented in various ways. However, json.dumps will just do a `str(<object>)` if
```

Comment:
should we just start using dict everywhere so we don't have to worry about stuff like this?